modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/OptionTTest.kt

```diff
@@ -102,47 +97,50 @@ class OptionTTest : UnitSpec() {
         }
     }
 
+    fun nestedGenk() = object : GenK<Nested<OptionTPartialOf<ForId>, OptionTPartialOf<ForNonEmptyList>>> {
+      override fun <A> genK(gen: Gen<A>): Gen<Kind<Nested<OptionTPartialOf<ForId>, OptionTPartialOf<ForNonEmptyList>>, A>> =
+        gen.map {
+          OptionT.just(Id.monad(), OptionT.just(NonEmptyList.monad(), it)).nest()
```

Comment:
This never generates `None` cases on neither the outer layer of option nor on the inner layer, so this should change a bit

Reply:
yes i was aware of that. i was going to ask if you could look into this one. i was not able to come up with a proper one.

Reply:
I think there should be a extension method `GenK<F>.nested(gen2: GenK<G>): GenK<Nested<F, G>>` it could be implemented as:
```kotlin
fun <F, G> GenK<F>.nested(gen2: GenK<G>): GenK<Nested<F, G>> = object: GenK<Nested<F, G>> {
  override fun <A> genK(gen: Gen<A>): Gen<Kind<Nested<F, G>, A>> = genK(gen2.genK(gen)).map { it.nest() }
}
```
Wrote this here without an editor, so might contain typos and might not even work, just an idea.

If it does not work, I'll clone your repo and actually try something in the editor :see_no_evil:

Reply:
awesome!

just needed one tiny mod (the `this@nested`):
```
fun <F, G> GenK<F>.nested(GENKG: GenK<G>): GenK<Nested<F, G>> = object : GenK<Nested<F, G>> {
  override fun <A> genK(gen: Gen<A>): Gen<Kind<Nested<F, G>, A>> =
    this@nested.genK(GENKG.genK(gen)).map { it.nest() }
}
```

I'll put it into GenK for general use

Reply:
cool, based on this i could also create a `EqK.nested`

Reply:
tried to add it directly to `EqK`  but  `Nested` is not available there. could this go into the package `arrow.mtl.extensions` instead?

Reply:
Hmm too bad :/ Yeah I think adding it in in `arrow.mtl.typeclasses.Composed` is a good idea